juniper_warp/src/lib.rs

```diff
@@ -0,0 +1,363 @@
+/*!
+
+# juniper_warp
+
+This repository contains the [warp][warp] web server integration for
+[Juniper][Juniper], a [GraphQL][GraphQL] implementation for Rust.
+
+## Documentation
+
+For documentation, including guides and examples, check out [Juniper][Juniper].
+
+A basic usage example can also be found in the [Api documentation][documentation].
+
+## Examples
+
+Check [examples/warp_server][example] for example code of a working warp
+server with GraphQL handlers.
+
+## Links
+
+* [Juniper][Juniper]
+* [Api Reference][documentation]
+* [warp][warp]
+
+## License
+
+This project is under the BSD-2 license.
+
+Check the LICENSE file for details.
+
+[warp]: https://github.com/seanmonstar/warp
+[Juniper]: https://github.com/graphql-rust/juniper
+[GraphQL]: http://graphql.org
+[documentation]: https://docs.rs/juniper_warp
+[example]: https://github.com/graphql-rust/juniper_warp/blob/master/examples/warp_server
+
+*/
+
+#![deny(missing_docs)]
+#![deny(warnings)]
+
+#[macro_use]
+extern crate failure;
+extern crate juniper;
+extern crate serde_json;
+extern crate warp;
+
+#[cfg(test)]
+extern crate percent_encoding;
+
+use std::sync::Arc;
+use warp::{filters::BoxedFilter, Filter};
+
+/// Make a filter for graphql endpoint.
+///
+/// The `schema` argument is your juniper schema.
+///
+/// The `context_extractor` argument should be a filter that provides the GraphQL context required by the schema.
+///
+/// Example:
+///
+/// ```
+/// # extern crate juniper_warp;
+/// # #[macro_use]
+/// # extern crate juniper;
+/// # extern crate warp;
+/// #
+/// # use std::sync::Arc;
+/// # use warp::Filter;
+/// # use juniper::{EmptyMutation, RootNode};
+/// # use juniper_warp::make_graphql_filter;
+/// #
+/// # fn main() {
+/// type UserId = String;
+/// # #[derive(Debug)]
+/// struct AppState(Vec<i64>);
+/// struct ExampleContext(Arc<AppState>, UserId);
+///
+/// struct QueryRoot;
+///
+/// graphql_object! (QueryRoot: ExampleContext |&self| {
+///     field say_hello(&executor) -> String {
+///         let context = executor.context();
+///
+///         format!("good morning {}, the app state is {:?}", context.1, context.0)
+///     }
+/// });
+///
+/// let schema = RootNode::new(QueryRoot, EmptyMutation::new());
+///
+/// let app_state = Arc::new(AppState(vec![3, 4, 5]));
+/// let app_state = warp::any().map(move || app_state.clone());
+///
+/// let context_extractor = warp::any()
+///     .and(warp::header::<String>("authorization"))
+///     .and(app_state)
+///     .map(|auth_header: String, app_state: Arc<AppState>| {
+///         let user_id = auth_header; // we believe them
+///         ExampleContext(app_state, user_id)
+///     })
+///     .boxed();
+///
+/// let graphql_filter = make_graphql_filter(schema, context_extractor);
+///
+/// let graphql_endpoint = warp::path("graphql")
+///     .and(warp::post(graphql_filter));
```

Comment:
this style of method filter is going to be deprecated, I'll change this next time I touch the code

Reply:
👍